### PR TITLE
[SYSTEMML-1816] toString does not print negative 0s anymore

### PR DESCRIPTION
**Pull request by nakul02**
Ping @deroneriksson

### CODE REVIEW

**Comment by akchinSTC** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1827/

**Comment by mboehm7** [comment]
Both positive and negative zero are considered zero - our sparse representation would never store a negative zero as a non-zero. Accordingly, you can leave the code for sparse outputs unchanged (note that 0.0==-0.0 and hence the existing check value != 0.0 already discards -0.0 as well)

**Comment by nakul02** [comment]
@mboehm7 - this is only for consistency of printing out the values to `stdout`. According to the JIRA, `toString` should print out the matrix the same way that iterating over each element and using `as.scalar` would. Nothing more.

@deroneriksson - any thoughts?

**Comment by mboehm7** [comment]
@nakul02 I can see that - what I'm saying is that this PR includes unnecessary changes (except the dense conversion case). Since sparse blocks will never contain negative zeros, there is no point in handling it when converting a sparse block to string. Also checks such as "!value.equals(0.0d) && !value.equals(-0.0d)" are unnecessarily convoluted and should be replaced with "value != 0".

**Comment by nakul02** [comment]
@mboehm7 - fixed

**Comment by akchinSTC** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1836/<h2>Failed Tests: <span class='status-failure'>2</span></h2><h3><a name='SystemML-PullRequestBuilder/org.apache.systemml:systemml' /><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1836/org.apache.systemml$systemml/testReport'>SystemML-PullRequestBuilder/org.apache.systemml:systemml</a>: <span class='status-failure'>2</span></h3><ul><li><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1836/org.apache.systemml$systemml/testReport/org.apache.sysml.test.integration.functions.data/FullReblockTest/testBinaryBlockSingeMSparseMR/'><strong>org.apache.sysml.test.integration.functions.data.FullReblockTest.testBinaryBlockSingeMSparseMR</strong></a></li><li><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1836/org.apache.systemml$systemml/testReport/org.apache.sysml.test.integration.functions.frame/FrameMatrixReblockTest/testFrameWriteMultipleSparseBinarySpark/'><strong>org.apache.sysml.test.integration.functions.frame.FrameMatrixReblockTest.testFrameWriteMultipleSparseBinarySpark</strong></a></li></ul>

**Comment by deroneriksson** [comment]
Test this Jenkins

**Comment by akchinSTC** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1847/<h2>Failed Tests: <span class='status-failure'>1</span></h2><h3><a name='SystemML-PullRequestBuilder/org.apache.systemml:systemml' /><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1847/org.apache.systemml$systemml/testReport'>SystemML-PullRequestBuilder/org.apache.systemml:systemml</a>: <span class='status-failure'>1</span></h3><ul><li><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1847/org.apache.systemml$systemml/testReport/org.apache.sysml.test.integration.functions.data/FullReblockTest/testTextCellMultipleMSparseMR/'><strong>org.apache.sysml.test.integration.functions.data.FullReblockTest.testTextCellMultipleMSparseMR</strong></a></li></ul>

File: src/main/java/org/apache/sysml/runtime/util/DataConverter.java [hunk 1]
```diff
@@ -840,7 +840,9 @@ public static String toString(MatrixBlock mb, boolean sparse, String separator,
 					IJV ijv = sbi.next();
 					int row = ijv.getI();
 					int col = ijv.getJ();
-					double value = ijv.getV();
+					Double value = ijv.getV();
+					if (value.equals(-0.0d))
+						value = 0.0;
```
Comment (mboehm7) [inline src/main/java/org/apache/sysml/runtime/util/DataConverter.java]: please, revert this.

File: src/main/java/org/apache/sysml/runtime/util/DataConverter.java [hunk 2]
```diff
@@ -850,7 +852,7 @@ public static String toString(MatrixBlock mb, boolean sparse, String separator,
 			} else {	// Block is in dense format
 				for (int i=0; i<rowLength; i++){
 					for (int j=0; j<colLength; j++){
-						double value = mb.getValue(i, j);
+						Double value = mb.getValue(i, j);
```
Comment (mboehm7) [inline src/main/java/org/apache/sysml/runtime/util/DataConverter.java]: there is no need to convert this double to a Double object here.

**Comment by nakul02** [comment]
@mboehm7 - addressed your comments.

**Comment by mboehm7** [comment]
LGTM - thanks @nakul02

**Comment by akchinSTC** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1897/<h2>Failed Tests: <span class='status-failure'>1</span></h2><h3><a name='SystemML-PullRequestBuilder/org.apache.systemml:systemml' /><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1897/org.apache.systemml$systemml/testReport'>SystemML-PullRequestBuilder/org.apache.systemml:systemml</a>: <span class='status-failure'>1</span></h3><ul><li><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1897/org.apache.systemml$systemml/testReport/org.apache.sysml.test.integration.functions.parfor/ParForDataPartitionLeftIndexingTest/testParForDataPartitionLeftindexingRowSparseMR/'><strong>org.apache.sysml.test.integration.functions.parfor.ParForDataPartitionLeftIndexingTest.testParForDataPartitionLeftindexingRowSparseMR</strong></a></li></ul>

**Comment by nakul02** [comment]
Thanks @mboehm7, I shall merge.

**Pull request closed without merging**